contexts/network.tsx

```diff
@@ -67,42 +66,59 @@ export const NetworkProvider: React.FC = function ({ children }) {
   }, [query, activeNetwork]);
 
   const updateNetworkParameters = useCallback(() => {
-    if (!beproServiceStarted || activeNetwork?.councilAmount) return;
+    if (!DAOService || activeNetwork?.councilAmount || !activeNetwork?.networkAddress) return;
 
+    
     Promise.all([
-      BeproService.getNetworkParameter("councilAmount"),
-      BeproService.getNetworkParameter("disputableTime"),
-      BeproService.getNetworkParameter("draftTime"),
-      BeproService.getNetworkParameter("oracleExchangeRate"),
-      BeproService.getNetworkParameter("mergeCreatorFeeShare"),
-      BeproService.getNetworkParameter("percentageNeededForDispute")
-    ]).then(values => {
-      setActiveNetwork({
-        ...activeNetwork,
-        councilAmount: values[0],
-        disputableTime: values[1] / 1000,
-        draftTime: values[2] / 1000,
-        oracleExchangeRate: values[3],
-        mergeCreatorFeeShare: values[4],
-        percentageNeededForDispute: values[5]
+        DAOService.getNetworkParameter("councilAmount"),
+        DAOService.getNetworkParameter("disputableTime"),
+        DAOService.getNetworkParameter("draftTime"),
+        DAOService.getNetworkParameter("oracleExchangeRate"),
+        DAOService.getNetworkParameter("mergeCreatorFeeShare"),
+        DAOService.getNetworkParameter("proposerFeeShare"),
+        DAOService.getNetworkParameter("percentageNeededForDispute"),
+        DAOService.getTreasury()
+    ])
+      .then(([councilAmount, 
+              disputableTime, 
+              draftTime, 
+              oracleExchangeRate, 
+              mergeCreatorFeeShare,
+              proposerFeeShare,
+              percentageNeededForDispute, 
+              treasury]) => {
+        setActiveNetwork({
+          ...activeNetwork,
+          councilAmount,
+          disputableTime: disputableTime / 1000,
+          draftTime: draftTime / 1000,
+          oracleExchangeRate,
+          mergeCreatorFeeShare,
+          proposerFeeShare,
+          percentageNeededForDispute,
+          treasury
+        });
       });
-    });
-  }, [activeNetwork, beproServiceStarted]);
+  }, [activeNetwork, DAOService]);
 
   useEffect(() => {
     updateActiveNetwork();
-  }, [query]);
+  }, [query?.network]);
+
+  useEffect(() => {
+    if (activeNetwork) updateNetworkParameters();
+  }, [DAOService, activeNetwork]);
 
   useEffect(() => {
-    if (query?.network) updateNetworkParameters();
-  }, [beproServiceStarted, query?.network]);
+    if(activeNetwork?.networkAddress) changeNetwork(handleNetworkAddress(activeNetwork));
```

Comment:
should we attest that activeNetwork is not the activeNetwork ? (so we dont change into the same network, triggering the useEffect again?)